internal/smartlink/smartlink.go

```diff
@@ -131,15 +138,20 @@ func UnlinkContents(src, dest string) error {
 	return nil
 }
 
+func isSymlink(src string) bool {
+	target, err := fileutils.SymlinkTarget(src)
+	return err == nil && src != target
+}
+
```

Comment:
```suggestion
```

`src != target` does not assert whether something is a symlink. Is this a concern we have? We could add the error check and return an error, but this doesn't feel relevant to the issue at hand. ie. the bug we're addressing isn't that a file is linking to itself.